'duplicate symbols for architecture x86_64' mac os x php7 link error

Building the extension fails on mac os x 10.11.4 with error
`duplicate symbols for architecture x86_64`

Output of the build log:

```
/bin/sh /Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman/libtool --mode=compile cc  -I. -I/Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman -DPHP_ATOM_INC -I/Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman/include -I/Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman/main -I/Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/main -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/TSRM -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/Zend -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/ext -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/ext/date/lib -I/usr/local/include  -DHAVE_CONFIG_H  -g -O0 -Wall   -c /Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman/php_gearman_client.c -o php_gearman_client.lo
 cc -I. -I/Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman -DPHP_ATOM_INC -I/Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman/include -I/Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman/main -I/Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/main -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/TSRM -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/Zend -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/ext -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/ext/date/lib -I/usr/local/include -DHAVE_CONFIG_H -g -O0 -Wall -c /Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman/php_gearman_client.c  -fno-common -DPIC -o .libs/php_gearman_client.o
/bin/sh /Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman/libtool --mode=compile cc  -I. -I/Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman -DPHP_ATOM_INC -I/Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman/include -I/Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman/main -I/Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/main -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/TSRM -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/Zend -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/ext -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/ext/date/lib -I/usr/local/include  -DHAVE_CONFIG_H  -g -O0 -Wall   -c /Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman/php_gearman_task.c -o php_gearman_task.lo
 cc -I. -I/Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman -DPHP_ATOM_INC -I/Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman/include -I/Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman/main -I/Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/main -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/TSRM -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/Zend -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/ext -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/ext/date/lib -I/usr/local/include -DHAVE_CONFIG_H -g -O0 -Wall -c /Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman/php_gearman_task.c  -fno-common -DPIC -o .libs/php_gearman_task.o
/bin/sh /Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman/libtool --mode=link cc -DPHP_ATOM_INC -I/Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman/include -I/Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman/main -I/Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/main -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/TSRM -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/Zend -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/ext -I/Users/dbaltas/.phpbrew/php/php-7.0.3/include/php/ext/date/lib -I/usr/local/include  -DHAVE_CONFIG_H  -g -O0 -Wall   -o gearman.la -export-dynamic -avoid-version -prefer-pic -module -rpath /Users/dbaltas/.phpbrew/build/php-7.0.3/ext/gearman/modules  php_gearman.lo php_gearman_client.lo php_gearman_task.lo -Wl,-rpath,/usr/local/lib -L/usr/local/lib -lgearman -Wl,-rpath,/usr/local/lib -L/usr/local/lib -lgearman -Wl,-rpath,/usr/local/lib -L/usr/local/lib -lgearman -Wl,-rpath,/usr/local/lib -L/usr/local/lib -lgearman
cc ${wl}-flat_namespace ${wl}-undefined ${wl}suppress -o .libs/gearman.so -bundle  .libs/php_gearman.o .libs/php_gearman_client.o .libs/php_gearman_task.o  -L/usr/local/lib -lgearman  -Wl,-rpath -Wl,/usr/local/lib -Wl,-rpath -Wl,/usr/local/lib -Wl,-rpath -Wl,/usr/local/lib -Wl,-rpath -Wl,/usr/local/lib
duplicate symbol _gearman_client_obj_handlers in:
    .libs/php_gearman.o
    .libs/php_gearman_client.o
duplicate symbol _gearman_client_ce in:
    .libs/php_gearman.o
    .libs/php_gearman_client.o
duplicate symbol _gearman_exception_ce in:
    .libs/php_gearman.o
    .libs/php_gearman_client.o
duplicate symbol _gearman_client_obj_handlers in:
    .libs/php_gearman.o
    .libs/php_gearman_task.o
duplicate symbol _gearman_task_obj_handlers in:
    .libs/php_gearman.o
    .libs/php_gearman_task.o
duplicate symbol _gearman_client_ce in:
    .libs/php_gearman.o
    .libs/php_gearman_task.o
duplicate symbol _gearman_exception_ce in:
    .libs/php_gearman.o
    .libs/php_gearman_task.o
duplicate symbol _gearman_task_ce in:
    .libs/php_gearman.o
    .libs/php_gearman_task.o
ld: 8 duplicate symbols for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [gearman.la] Error 1
```

Running cc with `-v` flag produced the same errors including some system details that may help

```
15:56:07 /tmp/pecl-gearman (master)$ cc -v ${wl}-flat_namespace ${wl}-undefined ${wl}suppress -o .libs/gearman.so -bundle  .libs/php_gearman.o .libs/php_gearman_client.o .libs/php_gearman_task.o  -L/usr/local/lib -lgearman  -Wl,-rpath -Wl,/usr/local/lib -Wl,-rpath -Wl,/usr/local/lib -Wl,-rpath -Wl,/usr/local/lib -Wl,-rpath -Wl,/usr/local/lib
Apple LLVM version 7.3.0 (clang-703.0.31)
Target: x86_64-apple-darwin15.4.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
 "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld" -demangle -dynamic -arch x86_64 -bundle -flat_namespace -macosx_version_min 10.11.0 -undefined suppress -undefined suppress -o .libs/gearman.so -L/usr/local/lib .libs/php_gearman.o .libs/php_gearman_client.o .libs/php_gearman_task.o -lgearman -rpath /usr/local/lib -rpath /usr/local/lib -rpath /usr/local/lib -rpath /usr/local/lib -lSystem /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/7.3.0/lib/darwin/libclang_rt.osx.a
```

@wcgallego unfortunately the latest master (bbc426f5609269de13a0edbea0a821c0c0b952f3) increases the number of duplicated symbols on OS X

```
duplicate symbol _gearman_client_obj_handlers in:
    .libs/php_gearman.o
    .libs/php_gearman_client.o
duplicate symbol _gearman_task_obj_handlers in:
    .libs/php_gearman.o
    .libs/php_gearman_client.o
duplicate symbol _gearman_client_ce in:
    .libs/php_gearman.o
    .libs/php_gearman_client.o
duplicate symbol _gearman_exception_ce in:
    .libs/php_gearman.o
    .libs/php_gearman_client.o
duplicate symbol _gearman_task_ce in:
    .libs/php_gearman.o
    .libs/php_gearman_client.o
duplicate symbol _gearman_client_obj_handlers in:
    .libs/php_gearman.o
    .libs/php_gearman_worker.o
duplicate symbol _gearman_worker_obj_handlers in:
    .libs/php_gearman.o
    .libs/php_gearman_worker.o
duplicate symbol _gearman_job_obj_handlers in:
    .libs/php_gearman.o
    .libs/php_gearman_worker.o
duplicate symbol _gearman_client_ce in:
    .libs/php_gearman.o
    .libs/php_gearman_worker.o
duplicate symbol _gearman_worker_ce in:
    .libs/php_gearman.o
    .libs/php_gearman_worker.o
duplicate symbol _gearman_exception_ce in:
    .libs/php_gearman.o
    .libs/php_gearman_worker.o
duplicate symbol _gearman_job_ce in:
    .libs/php_gearman.o
    .libs/php_gearman_worker.o
duplicate symbol _gearman_job_obj_handlers in:
    .libs/php_gearman.o
    .libs/php_gearman_job.o
duplicate symbol _gearman_exception_ce in:
    .libs/php_gearman.o
    .libs/php_gearman_job.o
duplicate symbol _gearman_job_ce in:
    .libs/php_gearman.o
    .libs/php_gearman_job.o
duplicate symbol _gearman_client_obj_handlers in:
    .libs/php_gearman.o
    .libs/php_gearman_task.o
duplicate symbol _gearman_task_obj_handlers in:
    .libs/php_gearman.o
    .libs/php_gearman_task.o
duplicate symbol _gearman_client_ce in:
    .libs/php_gearman.o
    .libs/php_gearman_task.o
duplicate symbol _gearman_exception_ce in:
    .libs/php_gearman.o
    .libs/php_gearman_task.o
duplicate symbol _gearman_task_ce in:
    .libs/php_gearman.o
    .libs/php_gearman_task.o
ld: 20 duplicate symbols for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [gearman.la] Error 1
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,8 +1,16 @@
-sudo: required
-dist: trusty
 language: php
-php:
-    - '7.0'
+
+matrix:
+  include:
+    - os: linux
+      dist: trusty
+      sudo: required
+      php: 7.0
+    - os: osx
+      sudo: required
+      language: generic
+      env:
+         - _PHP: 'php70'
 
 addons:
   apt:
@@ -10,6 +18,15 @@ addons:
     - gearman-job-server
     - libgearman-dev
     - gearman-tools
+
+before_install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update ; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew tap homebrew/dupes ; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew tap homebrew/versions ; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew tap homebrew/homebrew-php ; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew install gearman ; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew services start gearman ; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew install $_PHP ; fi
 
 script:
     - phpize

--- a/php_gearman.c
+++ b/php_gearman.c
@@ -28,6 +28,16 @@
 #include <libgearman-1.0/interface/status.h>
 #include <libgearman-1.0/status.h>
 
+zend_class_entry *gearman_exception_ce;
+zend_class_entry *gearman_client_ce;
+zend_object_handlers gearman_client_obj_handlers;
+zend_class_entry *gearman_task_ce;
+zend_object_handlers gearman_task_obj_handlers;
+zend_class_entry *gearman_job_ce;
+zend_object_handlers gearman_job_obj_handlers;
+zend_class_entry *gearman_worker_ce;
+zend_object_handlers gearman_worker_obj_handlers;
+
 
 /* {{{ arginfo */
 ZEND_BEGIN_ARG_INFO_EX(arginfo_gearman_version, 0, 0, 0)

--- a/php_gearman.h
+++ b/php_gearman.h
@@ -33,7 +33,7 @@ typedef enum {
         GEARMAN_OBJ_CREATED = (1 << 0)
 } gearman_obj_flags_t;
 
-zend_class_entry *gearman_exception_ce;
+extern zend_class_entry *gearman_exception_ce;
 
 #define GEARMAN_EXCEPTION(__error, __error_code) { \
         zend_throw_exception(gearman_exception_ce, __error, __error_code); \

--- a/php_gearman_client.h
+++ b/php_gearman_client.h
@@ -25,8 +25,8 @@
 #include <libgearman-1.0/interface/status.h>
 #include <libgearman-1.0/status.h>
 
-zend_class_entry *gearman_client_ce;
-zend_object_handlers gearman_client_obj_handlers;
+extern zend_class_entry *gearman_client_ce;
+extern zend_object_handlers gearman_client_obj_handlers;
 
 zend_object *gearman_client_obj_new(zend_class_entry *ce);
 

--- a/php_gearman_job.h
+++ b/php_gearman_job.h
@@ -25,8 +25,8 @@
 #include <libgearman-1.0/interface/status.h>
 #include <libgearman-1.0/status.h>
 
-zend_class_entry *gearman_job_ce;
-zend_object_handlers gearman_job_obj_handlers;
+extern zend_class_entry *gearman_job_ce;
+extern zend_object_handlers gearman_job_obj_handlers;
 
 zend_object *gearman_job_obj_new(zend_class_entry *ce);
 

--- a/php_gearman_task.h
+++ b/php_gearman_task.h
@@ -27,8 +27,8 @@
 #include <libgearman-1.0/status.h>
 
 zend_object *gearman_task_obj_new(zend_class_entry *ce);
-zend_class_entry *gearman_task_ce;
-zend_object_handlers gearman_task_obj_handlers;
+extern zend_class_entry *gearman_task_ce;
+extern zend_object_handlers gearman_task_obj_handlers;
 
 typedef enum {
         GEARMAN_TASK_OBJ_CREATED = (1 << 0),

--- a/php_gearman_worker.h
+++ b/php_gearman_worker.h
@@ -27,8 +27,8 @@
 #include <libgearman-1.0/interface/status.h>
 #include <libgearman-1.0/status.h>
 
-zend_class_entry *gearman_worker_ce;
-zend_object_handlers gearman_worker_obj_handlers;
+extern zend_class_entry *gearman_worker_ce;
+extern zend_object_handlers gearman_worker_obj_handlers;
 
 zend_object *gearman_worker_obj_new(zend_class_entry *ce);
 

